### DISCUSSION TITLE
Enhikari V6 Plus (aka MAP-E) with Banana Pi R4

### DISCUSSION THREAD

**Discussion by kenneyd** [fakemanhk/openwrt-jp-ipoe#41] [General]
First of all thanks to @fakemanhk for sharing this awesome guide to get everything working with OpenWRT. I wanted to share my own experience with my hardware/ ISP.

I moved from Nuro's G10T plan (awful service, they give you a /64, they lock you in for three years, they're dropping dual stack in favor of MAP-E, and they won't let you use your own hardware with it), to enHikari's v6プラス plan. EnHikari gave me a 10G-EPON ONU, a /56 prefix address, and I temporarily rented their wifi router box, the NTT XG-100NE, to confirm my V6 prefix, test the connection, etc. 

For following this guide I used a BananaPi R4 I bought from aliexpress, that was set up with OpenWrt 24.10.0-rc3 (mainline version, not the one provided by BananaPi), and basically everything worked right away. I did have to enable the SLAAC based DHCP server so that android devices on my LAN would get their v6 address properly, which had the effect of the "Active DHCPv6 Leases" being empty (I think I saw this mentioned in another discussion as well).

My only hiccups were finding that the BPI-R4 doesn't support my 10GTek SFP+ module (the 10G-EPON unit uses RJ45 only), but luckily the R4 does support an older Xicom 10G module I had that runs significantly hotter. Then also finding that on the mainline OpenWRT I can't seem to enable hardware offloading (probably this would work in BananaPi's fork of OpenWRT?), which I think is limiting my throughput, as I can't get much above 2gbit/s for up or down on various speedtests sites now.

Anyways, I probably would never have figured all this out on my own, so big thanks for that👍

**Comment by kenneyd** [comment]
To add another note, there was a weird process when I returned the rental router where they took me offline for a few hours and changed some stuff on their end. Once it was finished my ipv6 prefix (and related ipv4 MAP-E address) handed out by enhikari DHCP server had changed slightly and I had to update the `/etc/config/network` file manually with the new info. I wonder if there's a way to have this file be updated automatically so that if the IP changes again in the future I don't have to manually update the config? 

Also, now that my LAN devices were getting assigned a v6 address some were preferring ipv6 over ipv4 for DNS, andmy pihole was getting bypassed, so I changed the"Announced IPV6 DNS Servers" to point to the new ipv6 address of my pihole instead of using the openwrt router for DNS announcements and everything started to get funneled through the pihole again 👍

**Comment by fakemanhk** [comment]
Thanks for the feedback, glad that my work helped here.

The BPI-R4 Linux mainline support is still work-in-progress, hardware acceleration is available according to Mediatek but driver isn't ready in mainline Linux kernel yet, you can actually try to follow up the OpenWrt [discussion thread](url=https://forum.openwrt.org/t/a-new-dual-10g-router-based-on-filogic-880-banana-pi-bpi-r4/163861) raised by me if you want (a very long thread).

And for Android with IPv6, we do not have option here since Google isn't putting any DHCPv6 support there so SLAAC is the only thing you can use here.

For your IPv6 prefix change, usually ISP doesn't change it suddenly, mine only changed once due to house move, so I haven't really think of getting any kind of automation to handle this.

PiHole on IPv6 network is somewhat challenging but yes your approach is correct, this is also what I am doing, previously with old NanoPi R2S/R4S I put PiHole LXC on my OpenWrt together so that I can have all-in-one solution.

**Comment by samqixin** [comment]
Thanks for the feedback on ISP choices. I'm moving to a house and considering NTT flet's or NURO. I should do some serious digging then.
